src/Tools/AI Test Toolkit/src/Logs/AITLogEntry.Codeunit.al

```diff
@@ -5,21 +5,163 @@
+    var
+        AITLogEntries: Record "AIT Log Entry";
+    begin
+        AITLogEntries.SetRange(Tag, Tag);
+        DrillDownFailedAITLogEntries(AITLogEntries, AITSuiteCode, LineNo);
+    end;
+
+    local procedure DrillDownFailedAITLogEntries(var AITLogEntries: Record "AIT Log Entry"; AITSuiteCode: Code[100]; LineNo: Integer)
+    var
         AITLogEntry: Page "AIT Log Entries";
     begin
         AITLogEntries.SetFilterForFailedTestProcedures();
         AITLogEntries.SetRange("Test Suite Code", AITSuiteCode);
-        AITLogEntries.SetRange(Version, VersionNo);
         if LineNo <> 0 then
             AITLogEntries.SetRange("Test Method Line No.", LineNo);
         AITLogEntry.SetTableView(AITLogEntries);
         AITLogEntry.Run();
     end;
+
+    procedure UpdateRunHistory(Code: Code[100]; LineNo: Integer; AITViewBy: Enum "AIT Run History - View By"; var TempAITRunHistory: Record "AIT Run History" temporary)
+    var
+        AITRunHistory: Record "AIT Run History";
+        SeenTags: List of [Text[20]];
+    begin
+        TempAITRunHistory.DeleteAll();
+        AITRunHistory.SetRange("Test Suite Code", Code);
+
+        if AITViewBy = AITViewBy::Version then
+            if AITRunHistory.FindSet() then
+                repeat
+                    TempAITRunHistory.TransferFields(AITRunHistory);
+                    TempAITRunHistory.Insert();
+                until AITRunHistory.Next() = 0;
+
+        if AITViewBy = AITViewBy::Tag then
+            if AITRunHistory.FindSet() then
+                repeat
+                    if not SeenTags.Contains(AITRunHistory.Tag) then begin
+                        TempAITRunHistory.TransferFields(AITRunHistory);
+                        TempAITRunHistory.Insert();
+                    end;
+                    SeenTags.Add(AITRunHistory.Tag);
+                until AITRunHistory.Next() = 0;
+
+        if (LineNo <> 0) then
+            TempAITRunHistory.SetRange("Line No. Filter", LineNo)
+    end;
+
+    procedure LookupTestMethodLine(TestSuiteCode: Code[100]; var LineNoFilter: Text; var LineNo: Integer)
+    var
+        AITTestMethodLine: Record "AIT Test Method Line";
+        AITTestMethodLines: Page "AIT Test Method Lines Lookup";
+    begin
+        AITTestMethodLine.SetRange("Test Suite Code", TestSuiteCode);
+
+        AITTestMethodLines.SetTableView(AITTestMethodLine);
+        AITTestMethodLines.LookupMode(true);
+
+        if not (AITTestMethodLines.RunModal() = Action::LookupOK) then
+            exit;
+
+        AITTestMethodLines.GetRecord(AITTestMethodLine);
+
+        AITTestMethodLine.CalcFields("Codeunit Name");
+        LineNoFilter := StrSubstNo(LineNoFilterLbl, AITTestMethodLine."Codeunit ID", AITTestMethodLine."Codeunit Name", AITTestMethodLine."Input Dataset");
+        LineNo := AITTestMethodLine."Line No.";
+    end;
+
+    procedure UpdateTestInput(TestInput: Text; TestInputView: Enum "AIT Test Input - View"): Text
```

Comment:
Same comment for the methods below, it seems like it should go to the different object.

Reply:
It's specifically related to viewing test data, which is stored in AIT Log Entry. It's not really relevant in another context. I think this is more accurate placement of the function than placing it in a generic management codeunit that already has way to wide of a responsibility?

Reply:
Moved to AIT Test Data codeunit.